api/storage_test.go

```diff
@@ -0,0 +1,86 @@
+package api
+
+import (
+	"context"
+	"errors"
+	log "github.com/sirupsen/logrus"
+	"io/fs"
+	"path"
+)
+
+//TestStorage implements a very basic mock style storage that utilizes context and a hashmap to drive all the seed data.
+// the purpose of this pattern is mainly ensuring that the expectations of the type of return from the various methods
+// will result in a valid run.
+
+// Aka ReadDir returns a list of file with only the basename, given that we should be able to find the appropriate file
+// using app convention and established Prefix in order to drive the application.  If any CRUD operation files, then
+// the test is consistent.
+type TestStorage struct {
+	m      map[string]interface{}
```

Comment:
Can this be simplified by making m a `map[string][]byte`?

Then there's no need for a top level "filedata" key.

Reply:
it might, but it's still unusred for now.  Likely will come into play to do some simple mocking.